vic/drivers/python/setup.py

```diff
@@ -86,6 +88,94 @@ def write_version_py(filename=None):
+def maybe_eval_string(s):
+    '''run eval on strings between brackets e.g. [20 + 3] becomes [23]'''
+    matches = re.findall("\[(.*?)\]", s)
+    for match in matches:
+        if match:
+            try:
+                s = s.replace(match, str(eval(match)))
+            except (NameError, SyntaxError):
+                pass
+    return s
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+def make_cffi_headers():
+    '''Process the C headers such that CFFI can interpret them'''
+
+    omissions = ['va_list',
+                 'error_print_atmos_energy_bal',
+                 'error_print_atmos_moist_bal',
+                 'error_print_canopy_energy_bal',
+                 'error_print_solve_T_profile',
+                 'error_print_surf_energy_bal',
+                 'ErrorPrintIcePackEnergyBalance',
+                 'ErrorPrintSnowPackEnergyBalance',
+                 'func_atmos_energy_bal',
+                 'func_atmos_moist_bal',
+                 'func_canopy_energy_bal',
+                 'func_surf_energy_bal',
+                 'IceEnergyBalance',
+                 'root_brent',
+                 'SnowPackEnergyBalance',
+                 'soil_thermal_eqn',
+                 'zwtvmoist_zwt',
+                 'zwtvmoist_moist']
+
+    args = ['gcc', '-std=c99', '-E',
+            '-P', os.path.join(vic_root_abs_path, 'vic', 'drivers',
+                               'python', 'src', 'globals.c'),
+            '-I%s' % os.path.join(vic_root_abs_path, 'vic', 'drivers',
+                                  'python', 'include', ''),
+            '-I%s' % os.path.join(vic_root_abs_path, 'vic', 'drivers',
+                                  'shared_all', 'include', ''),
+            '-I%s' % os.path.join(vic_root_abs_path, 'vic',
+                                  'vic_run', 'include', '')]
+
+    proc = subprocess.Popen(' '.join(args),
+                            shell=True,
+                            stderr=subprocess.PIPE,
+                            stdout=subprocess.PIPE)
+    retvals = proc.communicate()
+
+    stdout = retvals[0].decode()
+    stderr = retvals[1].decode()
+
+    if proc.returncode:
+        print(' '.join(args))
+        raise RuntimeError('preprocessor experienced an error: %s' % stderr)
+
+    with open('./vic_headers.py', 'w') as f:
+        # write python script header information
+        f.write('#!/usr/bin/env python\n')
+        f.write("'''\n    Preprocessed Headers for VIC Python Driver\n"
+                "    Last updated %s\n'''\n\n\n" % datetime.now())
+        # now we write the preprocessed headers, skipping the system headers
+        f.write("headers = '''\n")
+        skip_headers = True
+        for line in stdout.split('\n'):
+            # TODO: improve how we determine if we're done looping through
+            #       the system headers
```

Comment:
remove to-do statement - maybe the stdout could be loaded into a list and stored as a set for quick set queries (with string parsing).